src/store/actions.ts

```diff
@@ -299,5 +300,28 @@ export default {
 
   setRestorationExpiry ({ commit }, expiry: string) {
     commit('mutateRestorationExpiry', expiry)
+  },
+
+  setRestorationApprovalType ({ commit }, type: ApprovalTypes) {
+    commit('mutateRestorationApprovalType', type)
+  },
+
+  setStateFilingRestoration (context): void {
+    LegalServices.fetchFiling(context.getters.getStateFilingUrl)
+      .then((response) => {
+        context.commit('mutateStateFilingRestoration', response.restoration)
+      })
```

Comment:
This should a promise that is resolved on success and rejected on failure.

Example: https://github.com/bcgov/business-filings-ui/blob/f56ca3a2358bf3da406973e1fe74fd2225e20db8/src/store/Business/actions.ts#L8

Also, you should await the call to this action since it's async.

Also, the return type should be `Promise<any>`.

Reply:
Looks good 👍

Reply:
I see that `setStateFilingRestoration` is awaited in LimitedRestorationExtension.vue. Is the same code in the restoration conversion view?

Reply:
No it's not there. I didn't want to touch that file since it might be someone else's ticket. I can add it however. Should I do that?

Reply:
Yeah, we're working on both filings at once, so if you add something to one then you should update the other one too (if applicable, but I think it is here).

Reply:
It is applicable. Okay, noted. Will do that Sev.

Reply:
Looks good.